### PR TITLE
Log exception in ProcessImpl::run()

### PR DESCRIPTION
**Pull request by oschaaf**
Related to https://github.com/envoyproxy/nighthawk/issues/317

Log any exceptions and rethrow. This is to assist with
diagnosis once it happens again in the CI env.

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

### CODE REVIEW

File: source/client/process_impl.cc [hunk 1]
```diff
@@ -414,75 +414,82 @@ bool ProcessImpl::run(OutputCollector& collector) {
...
-  createBootstrapConfiguration(bootstrap, uris, request_source_uri, number_of_workers);
-  if (tracing_uri != nullptr) {
-    setupTracingImplementation(bootstrap, *tracing_uri);
-    addTracingCluster(bootstrap, *tracing_uri);
-  }
-  ENVOY_LOG(debug, "Computed configuration: {}", bootstrap.DebugString());
-  cluster_manager_ = cluster_manager_factory_->clusterManagerFromProto(bootstrap);
-  maybeCreateTracingDriver(bootstrap.tracing());
-  cluster_manager_->setInitializedCb([this]() -> void { init_manager_.initialize(init_watcher_); });
+  try {
```
Comment (mum4k) [inline source/client/process_impl.cc:417]: This was already a fairly long function to begin with. How hard would it be to refactor the content of the new try { } into a separate private method? That way we could separate the exception handling (the outer run method) from the business logic (the new private method) and make this a bit easier to follow.

Comment (oschaaf) [reply source/client/process_impl.cc:491]: Done in 9c89e74453724491aec1fbfe479a871545db6553

**Comment by oschaaf** [comment]
Coverage is flaking because of a single line sometimes not getting hit:

https://12132-180498819-gh.circle-artifacts.com/0/coverage/coverage/proc/self/cwd/source/client/client_worker_impl.cc.html#L55

https://12111-180498819-gh.circle-artifacts.com/0/coverage/coverage/proc/self/cwd/source/client/client_worker_impl.cc.html#L55

**Comment by mum4k** [comment]
Is the flaky coverage something we should address / track?

Also can you help me by triggering it again so this becomes able to merge?

**Comment by oschaaf** [comment]
> Is the flaky coverage something we should address / track?
> 
> Also can you help me by triggering it again so this becomes able to merge?

I'm trying to address this in-place, it should be possible to reliably hit the line I mentioned.

**Comment by oschaaf** [comment]
I punted the coverage flake to https://github.com/envoyproxy/nighthawk/issues/328
Also, added https://github.com/envoyproxy/nighthawk/pull/327/commits/e28e1851c168a3e2db7f7d5a4b4bfc4a299afc28 which has cleanup but should be a no-op. At this time I can't understand how the line I pointed out above could not be covered.

**Pull request merged**